scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -275,6 +308,11 @@ private void start(final InetSocketAddress bindAddress) throws IOException {
 		timer = Executors.newSingleThreadScheduledExecutor(
 				new DaemonThreadFactory("DTLS RetransmitTask-", NamedThreadFactory.SCANDIUM_THREAD_GROUP));
 
+		if (executor == null) {
+			// use a thread pool with max 6 * #CPU cores
+			executor = new StripedExecutorService(Runtime.getRuntime().availableProcessors() * 6);
```

Comment:
You were the one who required this in the first place because of your *remote DB PSK store*, right? ;-) My concern was more whether you think that the size is too small ...